Feature: Implement a new form group “author as institution”

Ali-GeoFZ opened the pull request:
### Implementierung einer neuen Formulargruppe Author Institution inkl. kompletter Backend- und Frontend-Integration 
Issue #607 

## Changes

- Neue Formulargruppe inkl. Eingabefelder, Validierung und UI-Integration implementiert
- EventHandler für dynamisches Hinzufügen/Löschen von Einträgen erstellt
- Backend-Funktionen zur Speicherung und Löschung erweitert
- Datenbankmodell (ER-Modell + Tabellen) angepasst
- XML-Export angepasst
- Lade- und Löschfunktionen ergänzt
- Vollständige Unit-Tests für JavaScript- und PHP-Logik hinzugefügt
- Barrierefreiheit geprüft und sichergestellt
- Dokumentation aktualisiert

## Notes for Reviewer
Bitte ausführlich testen, es könnte sein dass ich irgendwas vergessen hab :).

## Checklist
- [x] Mein Code folgt dem Style Guide.
- [x] Ich habe selbst eine Code Review durchgeführt.
- [x] Ich habe Kommentare zu schwer verständlichem Code hinzugefügt.
- [x] Ich habe PHP-Code nach PHPDoc-Standard dokumentiert bzw. die Code-Dokumentation angepasst.
- [x] Ich habe JavaScript-Code nach JSDoc-Standard dokumentiert bzw. die Code-Dokumentation angepasst.
- [x] Ich habe, wenn nötig, entsprechende Änderungen im ELMO Guide vorgenommen.
- [x] Ich habe, wenn nötig, entsprechende Änderungen in der ReadMe vorgenommen.
- [ ] Ich habe, wenn nötig, entsprechende Änderungen in der API-Dokumentation vorgenommen.
- [x] Falls ich ein neues Feature implementiert oder einen Bug gefixt habe, wurde der Changelog erweitert
- [x] Meine Änderungen erzeugen keine neuen Warnungen in der Konsole des Testbrowsers
- [x] Ich habe Unit-Tests hinzugefügt, die meinen Code abdecken
- [x] Neue und bereits vorhandene Unit-Tests werden lokal bestanden
- [x] Neue und bereits vorhandende automatische Unit-Tests werden im Pull Request bestanden
- [ ] Ich habe, wenn nötig die Selenium-Tests aktualisiert und ggf. neue hinzugefügt
- [ ] Neue und bereits vorhandene automatisierte Selenium-Tests werden im Pull Request bestanden
- [x] Ich habe sicher gestellt, dass die Änderungen den Barrierefreiheitsrichtlinien entsprechen.


## Known Issues
**KEINE**

McNamara84 commented:
Starting with the review now.

Daetha commented:
## Dear Ali,

Sorry if you had to wait - I got a bit stuck with the review.

First of all, I want to encourage you: building a new form group in our established project means that you need to consider 1003435583 details : )

So no worries, we will test your code as long as it takes.

## Review Findings

Your recent commits are on point, but there's still an issue with front-end validation. Currently, users cannot save or submit. The problem appears to be in `validateAuthorInstitutionRequirements`. 

**Testing approach**: I deleted this condition and everything worked again, so something should be wrong with it.

## Specific Issue Identified

Looking at the function, I didn't immediately spot the bug. However, this line might be problematic:

`var isauthorinstitutionAffiliationFilled = (fields.authorinstitutionAffiliation.val() && fields.authorinstitutionAffiliation.val().trim() !== '')`

**Suggested refactor**:
```
var affVal = (fields.authorinstitutionAffiliation.val() || '').trim();
var isauthorinstitutionAffiliationFilled = affVal !== '';
```
## UX Consideration

The meaning of the two fields "Organisation name" and "Organisation affiliation" was unclear to me as a beginner. Maybe the **help section** could provide more detail on what to write there. 

For example: Affiliation is more general (like GFZ) and name is more specific (like section 1.4). In this case, what should the validation logic do?

Best,
Sasha

Daetha reviewed (changes requested):
Please take a closer look on validation part

Ali-GeoFZ commented:
> ## Dear Ali,
> Sorry if you had to wait - I got a bit stuck with the review.
> 
> First of all, I want to encourage you: building a new form group in our established project means that you need to consider 1003435583 details : )
> 
> So no worries, we will test your code as long as it takes.
> 
> ## Review Findings
> Your recent commits are on point, but there's still an issue with front-end validation. Currently, users cannot save or submit. The problem appears to be in `validateAuthorInstitutionRequirements`.
> 
> **Testing approach**: I deleted this condition and everything worked again, so something should be wrong with it.
> 
> ## Specific Issue Identified
> Looking at the function, I didn't immediately spot the bug. However, this line might be problematic:
> 
> `var isauthorinstitutionAffiliationFilled = (fields.authorinstitutionAffiliation.val() && fields.authorinstitutionAffiliation.val().trim() !== '')`
> 
> **Suggested refactor**:
> 
> ```
> var affVal = (fields.authorinstitutionAffiliation.val() || '').trim();
> var isauthorinstitutionAffiliationFilled = affVal !== '';
> ```
> 
> ## UX Consideration
> The meaning of the two fields "Organisation name" and "Organisation affiliation" was unclear to me as a beginner. Maybe the **help section** could provide more detail on what to write there.
> 
> For example: Affiliation is more general (like GFZ) and name is more specific (like section 1.4). In this case, what should the validation logic do?
> 
> Best, Sasha



Dear Sasha,


Thank you for reviewing. I've done almost everything. As for the points I haven't implemented, here are the reasons why.


- What do you mean by "1003435583"? I didn’t quite catch that 🙂
- The problem with Submit and Save is due to the two form Groups:

**Data Sources**: Something seems to be wrong with this form group. Although there are no mandatory fields visible, some fields are still being treated as required.
**Technical**: The issue is with the field Error Handling Approach. This field is considered mandatory even when it is not visible, which blocks the form submission without clearly indicating which field is empty.

- Refactored  ✅
- In this branch, I haven’t added anything labeled as “Organization Name” or “Affiliation.” Perhaps you are referring to other form groups? If that’s the case, it would be helpful to create a detailed issue describing the problem.
<img width="1882" height="147" alt="Screenshot 2025-09-02 093546" src="https://github.com/user-attachments/assets/e2d00ae5-cb48-443e-ac86-c8dffb810e25" />

Daetha reviewed (approved):
Hi Ali,
Testing Results
Completed ✅

[x] Install.html generates db structure with test data
[x] Save button produces a valid xml
[x] DataCite Author Institutions are listed as Creators after Persons (good!)
[x] Load button includes information from xml to Person and Institution authors (much needed!)
[x] Clear button works
[x] Validation works: if Affiliation is provided, the name is required
[x] Help buttons work nicely
[x] Documentation is extended. Hopefully, K. will like it.
[x] The form group looks visually appealing

Consider Adding

[ ] invalid-feedback — red text under input field if input is empty or invalid
[ ] Front-end validation for name. Currently, numbers, special characters and spaces are allowed
[ ] $showAuthorInstitution = true variable to sample_settings
[ ] Drag-and-drop functionality to control the order of authors
[ ] Consider improving location of add and delete button on very slim screens

Conclusion
Great work, this is approved!
You can merge this branch and implement the improvements in a different PR.
Congratulations!
Sasha
